### PR TITLE
TPA-554: Fix Cohost harp proxy/manager config

### PR DESCRIPTION
**Pull request by JonathanRenon-EDB**
Ensure that both harp-proxy and harp-manager are correctly configured when both roles are cohosted on a node.

Use a different config file for proxy when cohost is detected. proxy-config.yml is used and service file override is set accordingly.

Fix non detected bug for harp service file override also for cohosted proxy
where override for manager would also be applied to proxy service.

Clean up unused vars set for harp config templating leftover from TPA-72 split of tasks

### CODE REVIEW

File: roles/harp/config/tasks/harp2.yml [hunk 1]
```diff
@@ -84,26 +80,46 @@
     and not( harp_http_options['cert_file'] is defined
           or harp_http_options['key_file'] is defined )
 
-- name: Install HARP configuration file
+- name: Detect cohost HARP proxy and manager
```
Comment (haroon-github) [inline roles/harp/config/tasks/harp2.yml:83]:
```suggestion
- name: Record whether HARP proxy and manager are co-hosted
```

File: roles/harp/service/tasks/main.yml [hunk 2]
```diff
@@ -9,6 +9,9 @@
 # The harp packages hardcode postgres/postgres, so we must override the
 # default unit definitions if we want to run harp services as another
 # user or group.
+- name: Detect cohost HARP proxy and manager
```
Comment (haroon-github) [inline roles/harp/service/tasks/main.yml:12]:
```suggestion
- name: Record whether HARP proxy and manager are co-hosted
```

File: roles/harp/service/templates/harp.service.j2 [hunk 3]
```diff
@@ -14,10 +14,14 @@ Group={{ postgres_user }}
 Restart=on-failure
 {%     endif %}
 {%   endif %}
-{% else %}
-{%   if postgres_user != 'postgres' or postgres_group != 'postgres' %}
+{% elif config_for=='proxy' %}
+{%   if postgres_user != 'postgres' or postgres_group != 'postgres' or harp_cohost %}
 [Service]
 User={{ postgres_user }}
 Group={{ postgres_user }}
```
Comment (haroon-github) [inline roles/harp/service/templates/harp.service.j2:21]:
```suggestion
Group={{ postgres_group }}
```

File: roles/harp/service/templates/harp.service.j2
Comment (haroon-github) [inline roles/harp/service/templates/harp.service.j2:1]: It won't let me comment/suggest becasue it's not part of the change, but there's `Group={{ postgres_user }}` config in under `[service]`, why doesn't it say `Group={{ postgres_group }}`?

**Review by amenonsen — APPROVED** [review]
Looks fine to me at a high level, feel free to merge if it works fine in testing.
